store/sqlstore/channel_store.go

```diff
@@ -2084,6 +2084,21 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string,
 		return nil, store.NewErrInvalidInput("Channel", "Id", fmt.Sprintf("%v", channelIds))
 	}
 
+	var threadsToUpdate []string
```

Comment:
The first half of this block should be before the Select query is executed, because for postgres this query is not only the select, is also the update, so you are not following the approach of using the "current existing LastViewAt" of the channel membership to detect the threads to mark as read.